remote/remote.go

```diff
@@ -93,6 +102,18 @@ func AddEmptyLayerOnSave() ImageOption {
 	}
 }
 
+// WithRegistry register a registry.
+func WithRegistry(repository string, insecure, insecureSkipVerify bool) ImageOption {
```

Comment:
```suggestion
// WithRegistrySetting registers options to use when accessing images in a registry in order to construct the image. The referenced images could include the base image, a previous image, or the image itself.
func WithRegistrySetting(repository string, insecure, insecureSkipVerify bool) ImageOption {
```

Reply:
I'm sorry if this is overly pedantic. It took me a bit of time to understand this code (why are there multiple registries per image? and so on...) and I feel like something like this would have helped.